### ISSUE TITLE
[Rule Request] Tuple Pattern Rule

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#2203]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Rule Request

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

From [Google's Swift Style Guide](https://google.github.io/swift/#tuple-patterns):

> Assigning variables through a tuple pattern (sometimes referred to as a tuple shuffle) is only permitted if the left-hand side of the assignment is unlabeled.

> Labels on the left-hand side closely resemble type annotations, and can lead to confusing code.

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

```swift
// ok
let (a, b) = (y: 4, x: 5.0)

// triggers
let (x: a, y: b) = (y: 4, x: 5.0)
let (x: Int, y: Double) = (y: 4, x: 5.0)
```

3. Should the rule be configurable, if so what parameters should be configurable?

Just severity.

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a
   rule as opt-in.

Probably enabled by default, depends on oss-check.
